modules/heat_conduction/src/interfacekernels/SideSetHeatTransferKernel.C

```diff
@@ -0,0 +1,202 @@
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "SideSetHeatTransferKernel.h"
+
+registerMooseObject("HeatConductionApp", SideSetHeatTransferKernel);
+
+template <>
+InputParameters
+validParams<SideSetHeatTransferKernel>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addClassDescription(
+      "Modeling conduction, convection, and radiation across internal side set.");
+  params.addParam<MaterialPropertyName>("conductance",
+                                        "gap_conductance",
+                                        "Conductivity of gap divided by effective gap width,"
+                                        "conductance ignored if not provided");
+  params.addCoupledVar("Tbulk_var", "Bulk temperature of gap as variable");
+  params.addParam<MaterialPropertyName>(
+      "Tbulk_mat", "gap_Tbulk", "Bulk temperature of gap as material");
+  params.addParam<MaterialPropertyName>(
+      "h_master",
+      "gap_h_master",
+      "Convective heat transfer coefficient (master face), convection ignored if not provided");
+  params.addParam<MaterialPropertyName>(
+      "h_neighbor",
+      "gap_h_neighbor",
+      "Convective heat transfer coefficient (neighbor face), convection ignored if not provided");
+  params.addParam<MaterialPropertyName>(
+      "emissivity_eff_master",
+      "gap_emissivity_eff_master",
+      "Effective emmissivity of master face, radiation ignored if not provided. "
+      "This value contains contributions from reflectivity, see SideSetHeatTransferMaterial "
+      "for details.");
+  params.addParam<MaterialPropertyName>(
+      "emissivity_eff_neighbor",
+      "gap_emissivity_eff_neighbor",
+      "Effective emmissivity of neighbor face, radiation ignored if not provided. "
+      "This value contains contributions from reflectivity, see SideSetHeatTransferMaterial "
+      "for details.");
+  return params;
+}
+
+SideSetHeatTransferKernel::SideSetHeatTransferKernel(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _cond(getMaterialProperty<Real>("conductance")),
+    _Tbulk_var(isParamValid("Tbulk_var") ? &coupledValue("Tbulk_var") : nullptr),
+    _Tbulk_mat(_Tbulk_var ? nullptr : &getMaterialProperty<Real>("Tbulk_mat")),
+    _hp(getMaterialProperty<Real>("h_master")),
+    _hm(getMaterialProperty<Real>("h_neighbor")),
+    _eps_p(getMaterialProperty<Real>("emissivity_eff_master")),
+    _eps_m(getMaterialProperty<Real>("emissivity_eff_neighbor"))
+{
+  if (parameters.isParamSetByUser("Tbulk_mat") && _Tbulk_var)
+    mooseError("Both Tbulk_mat and Tbulk_var set by user, cannot use both.");
```

Comment:
Minor: You might still consider using param error here. Just pick one parameter of the two. It'll point to that one in the input file when the user hits this error, which is still very helpful.